controller/pagecontroller.php

```diff
@@ -40,7 +40,18 @@ public function __construct($appName, IRequest $request, $userId, $cacheManager,
 	 */
 	public function index() {
 		$params = array('user' => $this -> userId,'devices'=>$this->locationManager->loadAll($this->userId));
-		return new TemplateResponse('maps', 'main', $params);
+		$response = new TemplateResponse('maps', 'main', $params);
+		if (class_exists('OCP\AppFramework\Http\ContentSecurityPolicy')) {
+			$csp = new \OCP\AppFramework\Http\ContentSecurityPolicy();
+			// map tiles
+			$csp->addAllowedImageDomain('http://*.mqcdn.com');
+			// marker icons
+			$csp->addAllowedImageDomain('https://api.tiles.mapbox.com');
```

Comment:
If you are sure that it are only these: You can keep them as-is. However, a lot of the protection offered by CSP is anyways gone due through https://github.com/owncloud/maps/pull/49#discussion_r36758549. – Shouldn't make things weaker.